### PR TITLE
Fix Interest.h to correctly pull from schema

### PR DESCRIPTION
**Pull request by JHuculak**
#### Description
Various places inside Interest.h were either using the wrong SDK functions or providing an incorrect index when trying to retrieve data from a bit of Worker_ComponentData. This PR fixes the following:
* Various places that were using `Schema_GetObjectCount` to check for presence of optional primitive types now use the appropriate `Schema_Get{Type}Count` for the type in question.
* Retrieving the query constraint for a query was invoking `Schema_IndexObject` with an index of 1; this is nonsensical for a value that is neither optional nor part of a collection. This invocation is now correctly made with an index of 0.

#### Release note
* Fixed a bug that prevented the Interest component from being initialized properly when provided with `Worker_ComponentData`.

#### Tests
Debug / I/O testing confirmed that Interest structs are now properly constructed & populated with expected values.

#### Documentation
Release note

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 **注意**：自虚幻引擎开发套件 v0.10.0 版本起，其日志提供中英文两个版本。每个日志的中文版本都置于英文版本之后。
 
 ## [`x.y.z`] - Unreleased
+* Fixed a bug that prevented the Interest component from being initialized properly when provided with `Worker_ComponentData`.
```
Comment (m-samiec) [inline CHANGELOG.md:11]: Please move this to the bug-fixes section

File: SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h [hunk 2]
```diff
@@ -325,20 +325,16 @@ inline QueryConstraint IndexQueryConstraintFromSchema(Schema_Object* Object, Sch
 		NewQueryConstraint.RelativeBoxConstraint->EdgeLength = GetCoordinateFromSchema(RelativeBoxConstraintObject, 1);
 	}
 
-	// option<int64> entity_id_constraint = 7;
-	if (Schema_GetObjectCount(QueryConstraintObject, 7) > 0)
+  //option<int64> entity_id_constraint = 7;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h:328]: formatting mix-up here

**Review by m-samiec — APPROVED** [review]
Thanks for the fix!

**Comment by JHuculak** [comment]
@m-samiec pinging for a quick reconfirmation on the review; had to add some stuff to make sure that TSchemaOption values were initialized properly when trying to index query constraints (apparently, writing to the contents of an uninitialized unique pointer is a bad thing to do...)

File: SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h [hunk 3]
```diff
@@ -279,6 +279,7 @@ inline QueryConstraint IndexQueryConstraintFromSchema(Schema_Object* Object, Sch
 	{
 		Schema_Object* SphereConstraintObject = Schema_GetObject(QueryConstraintObject, 1);
 
+		NewQueryConstraint.SphereConstraint = SphereConstraint{};
```
Comment (JHuculak) [inline SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h:282]: Done this way instead of via initializer list for readability.

**Review by m-samiec — APPROVED** [review]

**Pull request merged**